gtda/diagrams/features.py

```diff
@@ -494,3 +498,208 @@ def transform(self, X, y=None):
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        X = check_diagrams(X)
+        validate_params(self.get_params(), self._hyperparameters,
+                        exclude=['quantiser_params', 'n_jobs'])
+
+        self.homology_dimensions_ = sorted(set(X[0, :, 2]))
+        n_dimensions = len(self.homology_dimensions_)
+
+        quantiser_estimator = implemented_quantiser_recipes[self.quantiser]
+        self._weight_function = \
+            implemented_weight_recipes[self.weight_function]
+        self._contrast_function = \
+            implemented_contrast_recipes[self.contrast_function]
+
+        if isinstance(self.quantiser_params, list):
+            effective_quantiser_params_ = self.quantiser_params
+        else:
+            effective_quantiser_params_ = \
+                [self.quantiser_params] * n_dimensions
+
+        # Validation of quantiser_params
+        _hyperparameters = self._hyperparameters.copy()
+        _hyperparameters['quantiser_params']['of'] = \
+            _AVAILABLE_QUANTISERS[self.quantiser]
+        _hyperparameters = {
+            'quantiser_params': _hyperparameters['quantiser_params']
+        }
+
+        for quantiser_params in effective_quantiser_params_:
+            params = {'quantiser_params': quantiser_params}
+            validate_params(params, _hyperparameters)
+
+        n_clusters = [
+            effective_quantiser_params_[i]['n_clusters']
+            if 'n_clusters' in effective_quantiser_params_[i].keys()
+            else quantiser_estimator().n_clusters
+            for i in range(n_dimensions)
+        ]
+
+        self._cluster_centers = {}
+        self._inertias = {}
+        self.centers_ = np.empty(shape=(1, np.sum(n_clusters), 3))
+        n_points_start, n_points_end = 0, n_clusters[0]
+
+        for i, dim in enumerate(self.homology_dimensions_):
+            Xd_sub = _subdiagrams(X, [dim], remove_dim=True).reshape(-1, 2)
+            # Remove diagonal points
+            Xd_sub = Xd_sub[Xd_sub[:, 1] != Xd_sub[:, 0]].reshape(-1, 2)
+
+            sample_weight = self._weight_function(Xd_sub)
+            quantiser = quantiser_estimator(**effective_quantiser_params_[i])
+            quantiser.fit(X=Xd_sub, sample_weight=sample_weight)
+            self._cluster_centers[dim] = quantiser.cluster_centers_
+
+            if quantiser.n_clusters == 1:
+                dist_centers = pairwise_distances(Xd_sub)
+                np.fill_diagonal(dist_centers, 0)
+                self._inertias[dim] = np.array([np.max(dist_centers) / 2.])
+            else:
+                dist_centers = pairwise_distances(self._cluster_centers[dim])
+                dist_centers[dist_centers == 0] = np.inf
+                self._inertias[dim] = np.min(dist_centers, axis=0) / 2.
```

Comment:
If there is one cluster, we set the inertia to its "radius", which is half the maximal distance between the points belonging to that cluster (all of them, since there's only one).
If there are many clusters, the inertia is half of the minimal distance between the two.
Is my understanding correct?